baybe/searchspace/continuous.py

```
@@ -480,16 +495,73 @@ def _sample_from_polytope(
+                    for b in range(batch_size):
+                        # Repeat the constraint for each "row"
+                        botorch_tuple = (
+                            param_indices + b * n_params,
+                            coefficients,
+                            rhs,
+                        )
+                        if c.is_eq:
+                            eq_constraints.append(botorch_tuple)
+                        else:
+                            ineq_constraints.append(botorch_tuple)
+                else:
+                    # Get the indices of the parameters used in the constraint
+                    param_index = {
+                        name: self.parameter_names.index(name) for name in c.parameters
+                    }
+                    param_indices_list = [
+                        param_index[param] + b * n_params
+                        for param in c.parameters
+                        for b in range(batch_size)
+                    ]
+                    _, coefficients, rhs = c.to_botorch(
+                        parameters=self.parameters, batch_size=batch_size
+                    )
+                    botorch_tuple = (
+                        torch.tensor(param_indices_list),
+                        coefficients,
+                        rhs,
+                    )
+                    if c.is_eq:
+                        eq_constraints.append(botorch_tuple)
+                    else:
+                        ineq_constraints.append(botorch_tuple)
+            bounds_joint = torch.cat(
+                [torch.from_numpy(bounds) for _ in range(batch_size)], dim=-1
+            )
+
+            points = get_polytope_samples(
+                n=1,
+                bounds=bounds_joint,
+                equality_constraints=eq_constraints,
+                inequality_constraints=ineq_constraints,
+            )
+            points = points.reshape(batch_size, points.shape[-1] // batch_size)
```

why is this a `//` here?

---

Ok, I get this part, but then your approach is suboptimal. Because by using `//`, you are silently dropping potential error cases that could happen if the result of your computation is non-integer!
Just imagine that `points.shape[-1]` is 5 but `batch_size` is 2 --> silent bug.

And if you really didn't care about having an explicit specification of the remainder dimension, you could have just used `-1`, which auto-infers the size, which would have been much simpler. So your version is neither simple, nor logic-safe. 

So I'd recommend to use a `strict` conversion to int and then pass that instead.

---

Changed this in 0824606b1d1241eb169db4cc02ed0dc41a41b0c5 to make it more robust, please resolve if happy :)